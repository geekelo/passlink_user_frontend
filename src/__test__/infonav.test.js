import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter for testing

// Import InfoNav component
import InfoNav from '../homePage/javascripts/infonav';

test('renders InfoNav correctly', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <InfoNav />
    </MemoryRouter>,
  );

  // Initial render snapshot
  expect(asFragment()).toMatchSnapshot();
});

// Snapshot testing with react-test-renderer
test('InfoNav snapshot', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <InfoNav />
    </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
