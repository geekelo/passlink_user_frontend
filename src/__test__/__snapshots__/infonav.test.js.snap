// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders InfoNav correctly 1`] = `
<DocumentFragment>
  <div
    class="infoNav-section"
  >
    <div
      class="back-button"
    >
      <a
        aria-current="page"
        class="links infonav-icons active"
        href="/"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-house "
          data-icon="house"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
    <div
      class="back-button"
    >
      <div>
        <button
          aria-label="nav-open"
          class="infonav-btn opendropdown-icon"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bars infonav-icons"
            data-icon="bars"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          aria-label="nav-close"
          class="infonav-btn closedropdown-icon hide"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-xmark infonav-icons"
            data-icon="xmark"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        class="dropdown-section hide"
      >
        <a
          class="infonav-links links"
          href="/about"
        >
          About
        </a>
        <a
          class="infonav-links links"
          href="/contact"
        >
          Contact
        </a>
        <a
          class="infonav-links links"
          href="/features"
        >
          Features
        </a>
        <a
          class="infonav-links links"
          href="/privacy-policy"
        >
          Privacy Policy
        </a>
        <a
          class="infonav-links links"
          href="/faqs"
        >
          FAQs | Help
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
